# HG changeset patch
# User Rik <rik@octave.org>
# Date 1372640909 25200
#      Sun Jun 30 18:08:29 2013 -0700
# Node ID 1aebb613a5ac08d4c7ece9d041cd5a56e2a05f8e
# Parent  cc3743a91652c2a8e321abde8f3a31e8bfbce375
Rename re_read_readline_init_file to readline_re_read_init_file.


* libinterp/interpfcn/input.cc(Fre_read_readline_init_file): Rename function.

* NEWS: Announce renaming.

* doc/interpreter/basics.txi: Rename function in manual.

* libgui/src/resource-manager.cc: Rename function in list of functions.

* scripts/deprecated/re_read_readline_init_file.m: Deprecated m-file
to call readline_re_read_init_file.

* scripts/deprecated/module.mk: Add re_read_readline_init_file.m to
build system

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -173,26 +173,26 @@ Summary of important user-visible change
 
       helpdlg    listdlg   questdlg
       inputdlg   msgbox    warndlg
 
  ** Other new functions added in 3.8.0:
 
       base64_decode               ellipke         lines      
       base64_encode               erfcinv         polyeig                   
-      betaincinv                  erfi            rgbplot             
-      built_in_docstrings_file    expint          save_default_options  
-      cmpermute                   findfigs        shrinkfaces           
-      cmunique                    fminsearch      splinefit             
-      colorcube                   gallery         strjoin               
-      copyobj                     gco             struct2hdl            
-      dawson                      hdl2struct      tetramesh             
-      dblist                      history_save    waterfall           
-      doc_cache_create            importdata
-      ellipj                      iscolormap
+      betaincinv                  erfi            readline_re_read_init_file
+      built_in_docstrings_file    expint          readline_read_init_file 
+      cmpermute                   findfigs        rgbplot               
+      cmunique                    fminsearch      save_default_options  
+      colorcube                   gallery         shrinkfaces           
+      copyobj                     gco             splinefit             
+      dawson                      hdl2struct      strjoin               
+      dblist                      history_save    struct2hdl          
+      doc_cache_create            importdata      tetramesh           
+      ellipj                      iscolormap      waterfall           
 
  ** Deprecated functions.
 
     The following functions were deprecated in Octave 3.4 and have been
     removed from Octave 3.8.
                                            
       autocor    dispatch              is_global    setstr
       autocov    fstat                 krylovb      strerror
@@ -210,16 +210,18 @@ Summary of important user-visible change
       java_debug
       java_get
       java_invoke
       java_new
       java_set
       java_unsigned_conversion
       javafields
       javamethods
+      re_read_readline_init_file
+      read_readline_init_file
       saving_history
 
     The following keywords have been deprecated in Octave 3.8 and will
     be removed from Octave 3.12 (or whatever version is the second major
     release after 3.8):
 
       static
 
diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -746,17 +746,17 @@ a configuration file of commands (See th
 exact command syntax).  The default configuration file is normally
 @file{~/.inputrc}.
 
 Octave provides two commands for initializing Readline and thereby changing
 the command line behavior.
 
 @DOCSTRING(readline_read_init_file)
 
-@DOCSTRING(re_read_readline_init_file)
+@DOCSTRING(readline_re_read_init_file)
 
 @node Customizing the Prompt
 @subsection Customizing the Prompt
 @cindex prompt customization
 @cindex customizing the prompt
 
 The following variables are available for customizing the appearance of
 the command-line prompts.  Octave allows the prompt to be customized by
diff --git a/libgui/src/resource-manager.cc b/libgui/src/resource-manager.cc
--- a/libgui/src/resource-manager.cc
+++ b/libgui/src/resource-manager.cc
@@ -1385,16 +1385,17 @@ resource_manager::octave_keywords (void)
       "rank "
       "ranks "
       "rat "
       "rats "
       "rcond "
       "rdivide "
       "re_read_readline_init_file "
       "readdir "
+      "readline_re_read_init_file "
       "readline_read_init_file "
       "readlink "
       "real "
       "reallog "
       "realmax "
       "realmin "
       "realpow "
       "realsqrt "
diff --git a/libinterp/interpfcn/input.cc b/libinterp/interpfcn/input.cc
--- a/libinterp/interpfcn/input.cc
+++ b/libinterp/interpfcn/input.cc
@@ -1069,16 +1069,17 @@ DEFUN (readline_read_init_file, args, ,
   "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} readline_read_init_file (@var{file})\n\
 Read the readline library initialization file @var{file}.  If\n\
 @var{file} is omitted, read the default initialization file (normally\n\
 @file{~/.inputrc}).\n\
 \n\
 @xref{Readline Init File, , , readline, GNU Readline Library},\n\
 for details.\n\
+@seealso{readline_re_read_init_file}\n\
 @end deftypefn")
 {
   octave_value_list retval;
 
   int nargin = args.length ();
 
   if (nargin == 0)
     command_editor::read_init_file ();
@@ -1090,19 +1091,19 @@ for details.\n\
         command_editor::read_init_file (file);
     }
   else
     print_usage ();
 
   return retval;
 }
 
-DEFUN (re_read_readline_init_file, args, ,
+DEFUN (readline_re_read_init_file, args, ,
   "-*- texinfo -*-\n\
-@deftypefn {Built-in Function} {} re_read_readline_init_file ()\n\
+@deftypefn {Built-in Function} {} readline_re_read_init_file ()\n\
 Re-read the last readline library initialization file that was read.\n\
 @xref{Readline Init File, , , readline, GNU Readline Library},\n\
 for details.\n\
 @seealso{readline_read_init_file}\n\
 @end deftypefn")
 {
   octave_value_list retval;
 
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -14,16 +14,17 @@ deprecated_FCN_FILES = \
   deprecated/java_get.m \
   deprecated/java_invoke.m \
   deprecated/java_new.m \
   deprecated/java_unsigned_conversion.m \
   deprecated/java_set.m \
   deprecated/javafields.m \
   deprecated/javamethods.m \
   deprecated/polyderiv.m \
+  deprecated/re_read_readline_init_file.m \
   deprecated/read_readline_init_file.m \
   deprecated/saving_history.m \
   deprecated/shell_cmd.m \
   deprecated/studentize.m \
   deprecated/sylvester_matrix.m
 
 FCN_FILES += $(deprecated_FCN_FILES)
 
diff --git a/scripts/deprecated/re_read_readline_init_file.m b/scripts/deprecated/re_read_readline_init_file.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/re_read_readline_init_file.m
@@ -0,0 +1,40 @@
+## Copyright (C) 2013 Rik Wehbring
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Built-in Function} {} re_read_readline_init_file (@var{file})
+## This function has been deprecated.  Use
+## @code{@file{readline_re_read_init_file}} instead.
+## @seealso{readline_read_init_file}
+## @end deftypefn
+
+## Deprecated in 3.8
+
+function re_read_readline_init_file (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "re_read_readline_init_file is obsolete and will be removed from a future version of Octave, please use readline_re_read_init_file instead");
+  endif
+
+  readline_re_read_init_file (varargin{:});
+
+endfunction
+
